GomokuClient/packages/gomoku-core/src/hooks/useGameSession.ts

```diff
@@ -1,20 +1,34 @@
 import { useEffect } from "react";
 import { TileColor, useTiles } from "@/hooks/useTiles";
-import { typedStorage } from "@/shared/lib/utils";
+import { getDefaultHeaders } from "@/shared/lib/utils";
 import { useSignalRConnection } from "@/context";
+import { SwaggerServices } from "@/api";
+import { genToArray } from "@/utils/gen.utility";
 
-export const useJoinGame = (gameID: string) => {
-  const { tiles, winner, addTile, lastTile } = useTiles(gameID);
+export const useGameSession = (gameID: string) => {
+  const { tiles, setTiles, winner, addTile, lastTile, setLastTile } =
+    useTiles();
   const { hubProxy, isConnected, registerEventHandlers } =
     useSignalRConnection();
 
+  const getGameHistory = () => {
+    SwaggerServices.getApiGameByGameIdHistory({
+      headers: getDefaultHeaders(),
+      path: { gameId: gameID },
+    })
+      .then((response) => {
+        if (!response.data) {
+          throw new Error("Game history not received!");
+        }
+
+        setTiles(genToArray(response.data.gen));
+        setLastTile(response.data.movesHistory[response.data.movesCount]);
+      })
+      .catch((error) => console.warn(error));
+  };
+
   useEffect(() => {
-    const previousGameID = typedStorage.getItem("currentGameID");
-    if (previousGameID && previousGameID !== gameID) {
-      typedStorage.removeItem(`gameBoard_${previousGameID}`);
-      typedStorage.removeItem(`nextTurn_${previousGameID}`);
-    }
-    typedStorage.setItem("currentGameID", gameID);
+    getGameHistory();
   }, [gameID]);
```

Comment:
useEffect has a missing dependency: 'getGameHistory'. Either include it or remove the dependency array